tools/gfx/d3d12/render-d3d12.cpp

```diff
@@ -2308,7 +2336,7 @@ class D3D12Device : public RendererBase
 
             SLANG_ASSERT(srcSize <= destSize);
 
-            _uploadBufferData(encoder->m_d3dCmdList, buffer, offset, srcSize, src);
+            _uploadBufferData(encoder->m_device, encoder->m_d3dCmdList, buffer, offset, srcSize, src);
```

Comment:
(cont. previous comment) You can get the transient heap object here via `encoder->m_transientHeap`.